frontend/ui/widget/textboxwidget.lua

```diff
@@ -154,6 +165,41 @@ function TextBoxWidget:init()
         self.text_height = self.lines_per_page * self.line_height_px
     end
 
+    -- Check for Poor Text Formatting
+    if not self.charlist and self.text and type(self.text) == "string"
+            and self.text:sub(1, #TextBoxWidget.PTF_HEADER) == TextBoxWidget.PTF_HEADER then
+        -- Support for very simple text styling (bold only for now)
+        self._ptf_char_is_bold = {}
+        -- Alas, we can't let any of our flag characters be fed to xtext (even with ASCII control
+        -- chars, it would give them a width, which would result at best in spurious added spacing).
+        -- So, split text into a table of chars, filter our flags out keeping track of where they
+        -- start and end bold, and rebuild self.text without them.
+        local charlist = util.splitToChars(self.text)
+        table.remove(charlist, 1)
+        local is_bold = false
+        local len = #charlist
+        local i = 1
+        while i <= len do
+            local ch = charlist[i]
+            if ch == TextBoxWidget.PTF_BOLD_START then
+                is_bold = true
+                table.remove(charlist, i)
+                len = len - 1
+            elseif ch == TextBoxWidget.PTF_BOLD_END then
+                is_bold = false
+                table.remove(charlist, i)
+                len = len - 1
+            else
+                if is_bold then
+                    self._ptf_char_is_bold[i] = true
+                end
+                i = i + 1
+            end
+        end
+        self.text = table.concat(charlist, "")
+        charlist = nil -- luacheck: no unused
```

Comment:
Dunno if/how this can accelerate GC of this possibly large table of chars vs. just going soon out of scope.
@NiLuJe : any thought?

Reply:
I'm all for it, yeah.

There's already *a lot* of churn going on with our text widgets, so if it's trivial to make intent clearer, I'm all for it ;).